### ISSUE TITLE
Sensitive variables not working for SSH targets

### ISSUE THREAD

**Issue by tothegills** [OctopusDeploy/Issues#3335]
Attempting to deploy to an SSH target while sensitive variables are set results in:

```
10:08:57   Error    |       Unable to parse sensitive-variables as valid JSON.
```

This was introduced by https://github.com/OctopusDeploy/OctopusDeploy/commit/16518163a9f29f04d18301e5988c54fc9b2900b4

The sensitive variable password is not being supplied as an argument to the bash script.

See: http://help.octopusdeploy.com/discussions/problems/52794
See: http://help.octopusdeploy.com/discussions/problems/52820-unable-to-parse-sensitive-variables-as-valid-json

**Comment by octoreleasebot** [comment]
Release Note: Fix sensitive variables for SSH target (introduced in 3.11.12)

**Comment by ekrapfl** [comment]
I am actually running across the same issue when using an offline package drop with sensitive variables.  I am getting the "unable to parse sensitive-variables as valid json" message, and I am never being prompted for a password.  Running Octopus 3.12.4, and it looks like Calamari version 3.7.51 from the offline package drop.

Thanks,
Eben

**Comment by ekrapfl** [comment]
I think I figured out the problem I was running across.  It wouldn't let me deploy the offline packages (from octopus) without putting a password on the deployment target (saying that I had sensitive variables... but I didn't).  Putting a password on the deployment target should have prompted me for that password when we ran the commands on the machine (but it didn't).  What I figured out is that even though I didn't have sensitive variables, there is one sensitive field in the process that I was using a non-sensitive variable to populate.  I think that confused the process somehow.

TLDR;
I marked the variables that go into the process as sensitive parameters as sensitive, so now it actually prompts me for a password, and all is well.

Thanks,
Eben
